core/os/dir_darwin.odin

```diff
@@ -14,11 +14,12 @@ read_dir :: proc(fd: Handle, n: int, allocator := context.allocator) -> (fi: []F
 
 	dirpath: string
 	dirpath, err = absolute_path_from_handle(fd)
-
 	if err != ERROR_NONE {
 		return
 	}
 
+	defer delete(dirpath)
+
```

Comment:
I'd move the `defer delete(dirpath)` to before the `err` check.

Reply:
I took a look at this and `absolute_path_from_handle` only allocates when there isn't an error. In the error case it returns ` "" `

Reply:
Yeah, looking at the Darwin implementation of `absolute_path_from_handle`, it appears to only allocate when there isn't an error (as thePHTest says).